packages/muze-axis/src/cartesian-axis/props.js

```diff
@@ -7,11 +8,34 @@ export const PROPS = {
     config: {
         sanitization: (context, value) => {
             const oldConfig = Object.assign({}, context._config || {});
-            value = mergeRecursive(oldConfig, value);
+            const mockedOldConfig = mergeRecursive({}, oldConfig);
+            value = mergeRecursive(mockedOldConfig, value);
+
             value.axisNamePadding = Math.max(value.axisNamePadding, 0);
-            if (value.orientation !== oldConfig.orientation) {
-                context.axis(context.createAxis(value));
+            const shouldAxesScaleUpdate = hasAxesConfigChanged(
+                value, oldConfig, ['interpolator', 'exponent', 'base', 'orientation']
+            );
+            const tickFormatter = context.getTickFormatter(value);
+
+            if (shouldAxesScaleUpdate) {
+                context._scale = context.createScale(value);
+                context._axis = context.createAxis(value);
             }
+            if (context._scale) {
+                const scale = context._scale;
+                const labelFunc = scale.ticks || scale.quantile || scale.domain;
+                context.sanitizedTickFormatter = tickFormatter ?
+                    tickFormatter(context.tickValues || context.axis().tickValues() || labelFunc()) :
+                    null;
+                // This method is used to set the padding between plots.
+                context.updateAxisPadding = (padding) => {
+                    if (typeof padding === 'number' && padding >= 0 && padding <= 1) {
+                        context._scale.padding(padding);
+                    }
+                };
+            }
+            context.fetchTickFormatter = tickFormatter;
```

Comment:
Why two formatters `fetchTickFormatter` and `sanitizedTickFormatter`?

Reply:
`sanitizedTickFormatter` is fed to d3 axis directly. It has been done here to avoid check during render in axis-renderer file
`fetchTickFormatter ` is used in getFormattedText method to apply tickformat and get the new text